libs/sdk-ui-dashboard/src/presentation/widget/insight/configuration/InsightAlertConfig/hooks/useAttributeValuesFromExecResults.ts

```diff
@@ -0,0 +1,105 @@
+// (C) 2019-2024 GoodData Corporation
+import { useCallback, useEffect, useState } from "react";
+import { DataViewFacade } from "@gooddata/sdk-ui";
+import {
+    areObjRefsEqual,
+    IAttributeDescriptor,
+    IAttributeMetadataObject,
+    isResultAttributeHeader,
+} from "@gooddata/sdk-model";
+
+import { IExecutionResultEnvelope } from "../../../../../../model/index.js";
+
+export type AttributeValue = {
+    title: string;
+    value: string;
+    name: string;
+};
+
+export function useAttributeValuesFromExecResults(execResult: IExecutionResultEnvelope | undefined) {
+    const [isResultLoading, setIsResultLoading] = useState(!execResult || execResult.isLoading);
+    const [dataView, setDataView] = useState<DataViewFacade | null>(null);
+
+    useEffect(() => {
+        if (execResult) {
+            execResult.executionResult?.readAll().then((data) => {
+                setDataView(DataViewFacade.for(data));
+                setIsResultLoading(false);
+            });
+        }
+    }, [execResult]);
+
+    const getAttributeValues = useCallback(
+        (attr: IAttributeMetadataObject): AttributeValue[] => {
+            const header = findHeader(dataView, attr);
+            const indexes = findDimIndexes(dataView, header);
+            return findAttributeValues(dataView, indexes);
+        },
+        [dataView],
+    );
+
+    return {
+        isResultLoading,
+        getAttributeValues,
+    };
+}
+
+function findHeader(dataView: DataViewFacade | null, attr: IAttributeMetadataObject) {
+    if (!dataView) {
+        return null;
+    }
+
+    return (
+        dataView
+            .meta()
+            .attributeDescriptors()
+            .find((descriptor) => {
+                return (
+                    areObjRefsEqual(descriptor.attributeHeader.ref, attr.ref) ||
+                    attr.displayForms.some((displayForm) => {
+                        return areObjRefsEqual(descriptor.attributeHeader.ref, displayForm.ref);
+                    })
+                );
+            }) ?? null
+    );
+}
+
+function findDimIndexes(
+    dataView: DataViewFacade | null,
+    header: IAttributeDescriptor | null,
+): [number, number] {
+    const def = [-1, -1] as [number, number];
```

Comment:
This would deserve some comments, for better understanding what is happening there 😁  (same applies to findAttributeValues)

Reply:
Added some description to function